packages/foam-core/src/markdown-provider.ts

```diff
@@ -143,45 +157,78 @@ export function createMarkdownReferences(
     return [];
   }
 
-  return graph
-    .getForwardLinks(noteId)
-    .map(link => {
-      let target = graph.getNote(link.to);
-      // if we don't find the target by ID we search the graph by slug
-      if (!target) {
-        const candidates = graph.getNotes({ slug: link.link.slug });
-        if (candidates.length > 1) {
+  // if note doesn't exist, we can't find its links
+  const note = graph.getNote(noteId);
+  if (!note) {
+    return [];
+  }
+
+  const forwardLinks = graph.getForwardLinks(noteId);
+
+  // Try to generate a definition for each [[link]] in the note.
+  //
+  // A note may have multiple [[link]] expressions to the same target
+  // note with different aliases.s
+  //
+  // - note.links contains all the [[link]] expressions
+  // - forwardLinks contains all the edges (links) between documents
+  return (
+    note.links
+      .map(linkExpression => {
+        // find the link between this and other document
+        const link = forwardLinks.find(
+          note => note.link.slug === linkExpression.slug
+        );
+
+        // if other document is not found, bail
+        if (!link) {
```

Comment:
looking at this I feel that variable should be called `connection`, as it would probably read  more easily and defines a separation between the link in the note and the connection in the graph.

```
if (!connection) {
```

(makes me wonder whether the incorrect naming is stemming from the `getForwardLinks` function, which we might instead wanna call `getForwardConnections`)

Reply:
Yeah, there definitely is some terminology overloading happening for "link" that we should disambiguate. I like Connection, but also Edge?

In either case, I don't think this is the right PR to make that change.

Reply:
yeah it's possible that `edge` is a better name, I suggested `connection` because that's what we use in `NoteGraph`, but I am easy about it.
I agree 100% we shouldn't make that refactoring as part of this PR